### PR TITLE
RetowerCEMC: Add optional masking for negative energy towers

### PR DESCRIPTION
**Pull request by Steepspace**
## Types of changes
[comment]: <> ( What types of changes does your code introduce? Put an `x` in all the boxes that apply: )
- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work for users)
- [ ] Requiring change in macros repository (Please provide links to the macros pull request in the last section)
- [x] I am a member of [GitHub organization of sPHENIX Collaboration](https://github.com/orgs/sPHENIX-Collaboration/people), EIC, or ECCE (contact Chris Pinkenburg to join)

## What kind of change does this PR introduce? (Bug fix, feature, ...)

Add a backward-compatible feature to optionally mask large negative energy EMCal towers if their energy falls below a specified threshold. Introduced two new setters to `RetowerCEMC` to configure the masking:
- `set_mask_negative_energy(bool)` (default: false)
- `set_negative_energy_threshold(double)` (default: -2.0 GeV)

## TODOs (if applicable)

[comment]: <> ( In case this is a draft PR, e.g. for running checks using Jenkins, please make the pull request as a draft: https://github.blog/2019-02-14-introducing-draft-pull-requests/  )

## Motivation / Context

This change adds optional masking for negative energy EMCal towers during retowering. The default behavior remains unchanged.

## Key Changes

- Added `set_mask_negative_energy(bool)`.
- Added `set_negative_energy_threshold(double)`.
- Marked negative energy towers as bad when masking is enabled and energy is below the threshold.
- Set defaults to disabled masking and `-2.0 GeV`.

## Potential Risk Areas

- Reconstruction behavior changes when the option is enabled.
- No I/O format changes are expected.
- No new thread-safety concerns are apparent.
- The performance impact should be negligible.

## Possible Future Improvements

- Add focused tests for threshold boundaries and masking behavior.
- Document the configuration in user-facing retowering guidance.
- Monitor reconstruction effects in representative datasets.

This summary is based on AI-generated change descriptions. AI can make mistakes, so contributors should verify the implementation and physics behavior before merging.

### CODE REVIEW

**Comment by Steepspace** [comment]
@coderabbitai review

**Comment by Steepspace** [comment]
@coderabbitai review

**Comment by Steepspace** [comment]
@coderabbitai review

File: offline/packages/jetbackground/RetowerCEMC.cc [hunk 1]
```diff
@@ -84,9 +84,15 @@ int RetowerCEMC::process_event(PHCompositeNode *topNode)
       unsigned int channelkey = towerinfosEM3->encode_key(channel);
       int ieta = towerinfosEM3->getTowerEtaBin(channelkey);
       int iphi = towerinfosEM3->getTowerPhiBin(channelkey);
-      rawtower_e[ieta][iphi] = tower->get_energy();
+      double energy = tower->get_energy();
+      rawtower_e[ieta][iphi] = energy;
       rawtower_time[ieta][iphi] = tower->get_time();
-      rawtower_status[ieta][iphi] = !tower->get_isGood();
+      bool is_bad = !tower->get_isGood();
+      if (_mask_negative_energy && energy < _negative_energy_threshold)
```
[the remark this answers is not in the document]
Comment (Steepspace) [reply offline/packages/jetbackground/RetowerCEMC.cc:91]: We use set_towerinfo(true) exclusively and the other mode is deprecated.

**Comment by Steepspace** [comment]
Superseded by https://github.com/sPHENIX-Collaboration/coresoftware/pull/4415
